libnavui-app/build.gradle

```diff
@@ -0,0 +1,53 @@
+apply plugin: 'com.android.library'
+apply plugin: 'kotlin-android'
+apply plugin: 'kotlin-kapt'
+apply plugin: 'kotlin-parcelize'
+apply plugin: 'com.jaredsburrows.license'
+apply plugin: 'com.mapbox.android.sdk.versions'
+apply from: "${rootDir}/gradle/ktlint.gradle"
+apply from: file("${rootDir}/gradle/artifact-settings.gradle")
+
+version = project.ext.versionName
+group = project.ext.mapboxArtifactGroupId
+
+android {
+    compileSdkVersion androidVersions.compileSdkVersion
+
+    compileOptions {
+        sourceCompatibility = JavaVersion.VERSION_1_8
+        targetCompatibility = JavaVersion.VERSION_1_8
+    }
+
+    defaultConfig {
+        minSdkVersion androidVersions.minSdkVersion
+        targetSdkVersion androidVersions.targetSdkVersion
+        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
+        consumerProguardFiles 'proguard-rules.pro', "${rootDir}/proguard/proguard-project.pro"
+
+        buildConfigField 'String', "MAPBOX_NAVIGATION_VERSION_NAME", String.format("\"%s\"", project.ext.versionName)
+    }
+
+    testOptions {
+        unitTests.returnDefaultValues = true
+        unitTests.includeAndroidResources = true
+    }
+}
+
+dependencies {
+    api project(":libnavui-base")
+    api project(":libnavui-voice")
```

Comment:
Maybe I am not wiring things here correctly and I need your help to properly set this up. 

The idea is to make `libnavui-app` an internal module that only `android-audio` and `drop-in` modules depend on.
The module will be used as a central place to put any common interfaces, models, and utilities relevant only to AA and Drop-In UI, and will be responsible for the creation and registration of shared SDK components (starting with MapboxAudioGuidance).

Reply:
Why would we want to make `libnavui-app` an internal module? If we do that, how do we handle the use case where a customer is building an app **without** using Drop-In UI but **wants** to add android-auto support. In that case the customer should be able to use `SharedApp` to have there app and android auto in sync.

Reply:
I would like to see what @LukasPaczos has to say about this. I think the new `libnavui-app` module is fine but we should not treat it to be internal. The new dependency tree would look like:
```
libnavui-app
     |__ libnavigation-android
```
- If you want to build your own custom turn-by-turn experience, using the complete feature set, add the following to your build.gradle.
```
api project(libnavigation-android)
```
- If you want to build a turn-by-turn experience, using Drop-In UI, add the following to your build.gradle.
```
api project(libnavui-dropin)
```
- If you want to build your own custom turn-by-turn experience along with android auto, using the complete feature set, add the following to your build.gradle.
```
api project(libnavui-app)
api project(libnavui-androidauto)
```
- If you want to build a turn-by-turn experience along with android auto, using Drop-In UI, add the following to your build.gradle.
```
api project(libnavui-dropin)
api project(libnavui-androidauto)
```

Reply:
This is one of the main complications of creating a separate module. Also because we're supporting different release chains. `libnavui-app` will need to be either, part of the SDK release chain (2.x.). Or a new release chain, like androidauto is.

### approach 1: `libnavui-app` is part of the 2.x

If `libnavui-app` is part of the 2.x release train then it can be bundled with drop-in-ui and ui-androidauto like this

```
libnavui-dropin
  -- api project(":libnavui-app")
libnavui-androidauto
  -- api implementation("com.mapbox.navigation:ui-apps:2.5.0-rc.1")
```

libnavui-androidauto needs to specify a version because it has a separate release train.

### approach 1: `libnavui-app` new chain 0.x.0

Another option is to make it a new sdk, where both libnavui-dropin and libnavui-androidauto depend on a specific version.
```
libnavui-app and libnavui-androidauto
  -- api implementation("com.mapbox.navigation:ui-apps:0.1.0")
```

### I'm preferring, approach 1: 

In my opinion, `ui-apps` does not include enough responsibility to justify a new release pipeline. So I'd vote for making it part of the 2.x pipeline.

Also, considering that `:libnavui-app` is mostly an internal artifact. I think we can declare it with `implementation` dependencies for what it specifically needs. `libnavui-dropin` and `libnavui-androidauto` do not need to get `libnavigation-android` from `libnavui-app`
```
libnavui-app
 -- implementation project(":libnavui-base")
 -- implementation project(":libnavui-voice")
libnavui-dropin
  -- api project(":libnavui-app")
libnavui-androidauto
  -- api implementation("com.mapbox.navigation:ui-apps:2.5.0-rc.1")
```

Reply:
All these complications. Also make me prefer a solution where we don't need to make a new module for it. Experimental APIs added to `libnavui-base` is still a reasonable solution. At least to start they can live there, if we find a new module necessary - we can come back to this.